homeassistant/components/notify/twitter.py

```diff
@@ -128,17 +140,63 @@ def upload_media_chunked(self, file, total_bytes, media_id):
         return media_id
 
     def upload_media_append(self, chunk, media_id, segment_id):
-        """Upload media, append phase."""
+        """Upload media, APPEND phase."""
         return self.api.request('media/upload',
                                 {'command': 'APPEND', 'media_id': media_id,
                                  'segment_index': segment_id},
                                 {'media': chunk})
 
     def upload_media_finalize(self, media_id):
-        """Upload media, finalize phase."""
+        """Upload media, FINALIZE phase."""
         return self.api.request('media/upload',
                                 {'command': 'FINALIZE', 'media_id': media_id})
 
+    def upload_media_check_status(self, media_id, processing_info=None):
+        """Upload media, STATUS phase."""
+        if processing_info is None:
+            return None
+
+        _LOGGER.debug("media processing %s status: %s", media_id,
+                      processing_info)
+
+        if processing_info['state'] == u'succeeded':
+            return True
+
+        if processing_info['state'] == u'failed':
+            _LOGGER.error("media processing failed")
+            return False
+
+        check_after_secs = processing_info['check_after_secs']
+        _LOGGER.debug("media processing waiting %s seconds to check status",
+                      str(check_after_secs))
+
+        time.sleep(check_after_secs)
```

Comment:
@pvizeli I'm not sure how to use `helpers/event.py`. Could you point me to an example in the hass codebase? (I'm coming up empty.)